weldx/geometry.py

```diff
@@ -1577,15 +1578,73 @@ def local_coordinate_system(
         orientation = tf.WXRotation.from_euler(
             "z", self._angle * relative_position * self._sign_winding
         ).as_matrix()
-        translation = np.array([0, -1, 0]) * self._radius * self._sign_winding
+        translation = np.array([0, -1, 0]) * self.radius * self._sign_winding
 
         coordinates = np.matmul(orientation, translation) - translation
         return tf.LocalCoordinateSystem(orientation, coordinates)
 
 
+class DynamicTraceSegment:
+    """Trace segment that can be defined by a ``SpatialSeries``."""
+
+    def __init__(self, series, max_s=1):
+        from weldx.core import SpatialSeries
+
+        self._series: SpatialSeries = series
+        self._max_s = max_s
+        self._length = self._len_expr() if series.is_expression else self._len_disc()
+
+    def _get_derivative(self, i):
+        me = self._series.data
+        exp = me.expression
+        # todo unit stripped -> how to proceed? how to cast all length units to mm?
+        subs = [(k, v[i].data.to_base_units().m) for k, v in me.parameters.items()]
+        return exp.subs(subs).diff("s")
```

Comment:
for performance it would probably be good to calculate the derivative (and other formulas needed) once after instantiation

Reply:
There is actually a lot of room for improvements, especially in the "quality of life" section. But I just wanted to get things running first. :)